foundations/javascript_basics/clean_code.md

```diff
@@ -42,63 +47,212 @@ const sumArray = function(array) {
-   ~~~
-   
-   ​
+~~~
 
-4.  Naming Things: Names for functions and variables should be descriptive.  Always use camelCase.  To keep things consistent and easy to read, variables should always begin with a noun or an adjective (that is, a noun phrase) and functions with a verb.  It is ok to use single characters as variable names in the context of a loop or a callback function, but not elsewhere.
+### Semicolons
 
-    ~~~javascript
-// Good
-    const numberOfThings = 10;
-    const myName = "Thor";
-    const selected = true;
+Semicolons are _mostly_ optional in JavaScript because the JS compiler will automatically insert them if they are omitted. This functionality CAN break in certain situations leading to bugs in your code so it is better to get used to adding semi-colons.
+
+Again: consistency is the main thing.
+
+### About comments
+
+Comments are a great tool. But like any good tool, they can be misused. Especially for someone early in their coding journey, it might be tempting to have comments that explain _everything_ the code is doing. This is not a good practice.
+
+Next we'll look into some common pitfalls in commenting and _why_ they are pitfalls.
+
+#### Don't comment when you should be using git
+
+It might be tempting to have comments in your code that explain the changes or additions you have made. For example:
+
+~~~javascript
+/**
+ * 2023-01-10: Removed unnecessary code that was causing confusion (RM)
+ * 2023-03-05: Simplified the code (JP)
+ * 2023-05-15: Removed functions that were causing bugs in production (LI)
+ * 2023-06-22: Added a new function to combine values (JR)
+ */
+~~~
+
+The problem is that you already have a tool to track changes - git! Keeping track of these comments will become a chore and you will have an incomplete picture of what has happened. Your files will also contain bloat that doesn't belong there.
+
+By using git, all this information will be neatly organized in the repository and readily accessible with `git log`.
+
+The same applies to code that is no longer used. If you need it again in the future, just turn to your git commits. Commenting out something while testing something else is of course ok but once a piece of code is not needed, just delete it. Don't have something like this hanging around in your files:
+
+~~~javascript
+theFunctionInUse();
+// oldFunction();
+// evenOlderUselessFunction();
+// whyAmIEvenHereImAncient():
+~~~
+
+#### Tell why, not how
+
+The purpose of comments is not to provide pseudo code that duplicates your code. Good comments explain the _reasons_ behind a piece of code.
+
+Let's look at an example to see this in practice:
+
+~~~javascript
+// Bad Example - comment doesn't tell why, only what and how
+
+// This function increments the value of i by 1
+function incrementI(i) {
+  i = i + 1; // Add one to i
+  return i;
+}
+
+// Better Example - comment tells a why
+
+// This function increments the value of index to move to the next element
```

Comment:
I'd love to include an example where a code comment is a great idea. Can we toss something like that in?